pkg/registry/cluster/storage/aggregate.go

```diff
@@ -0,0 +1,394 @@
+				impersonateToken, err := clusterImpersonateToken(ctx, cluster, secretGetter)
+				if err != nil {
+					klog.Errorf("failed to get impersonateToken for cluster %s: %v", cluster.Name, err)
+					return
+				}
+				statusCode, err := doClusterRequest(req.Method, requestURLStr(location.String(), proxyRequestInfo), transport, requester, impersonateToken)
+				if err != nil {
+					klog.Errorf("failed to do request for cluster %s: %v", cluster.Name, err)
+					return
+				}
+				if statusCode == http.StatusOK {
+					locker.Lock()
+					requestContexts = append(requestContexts, requestContext{
+						clusterName:      cluster.Name,
+						location:         location,
+						transport:        transport,
+						impersonateToken: impersonateToken,
+					})
+					locker.Unlock()
+				}
+			}(&cluster)
+		}
+		wg.Wait()
+		switch len(requestContexts) {
+		case 0:
+			http.Error(rw, "resource not found or don't have permission to get it", http.StatusNotFound)
+		case 1:
+			reqCtx := requestContexts[0]
+			setRequestHeader(req, requester, reqCtx.impersonateToken)
+			reqCtx.location.Path = path.Join(reqCtx.location.Path, proxyPath)
+			reqCtx.location.RawQuery = req.URL.RawQuery
+
+			handler := proxy.NewThrottledUpgradeAwareProxyHandler(reqCtx.location, reqCtx.transport,
+				true, false, responder)
+			handler.ServeHTTP(rw, req)
+		default:
+			clusterNames := make([]string, len(requestContexts))
+			for i, reqCtx := range requestContexts {
+				clusterNames[i] = reqCtx.clusterName
+			}
+			http.Error(rw, fmt.Sprintf("conflict resource, exist in more than one cluster: %s",
+				strings.Join(clusterNames, ",")), http.StatusConflict)
+		}
+	})
+}
+
+// nolint:gocyclo
+func requestWithoutResourceNameHandlerFunc(
+	ctx context.Context,
+	proxyPath string,
+	secretGetter func(context.Context, string, string) (*corev1.Secret, error),
+	clusterList *clusterapis.ClusterList,
+) http.Handler {
+	// For uncertain resource names in processing, we need to make further judgments
+	// based on the requested verb to determine whether to merge the request results
+	// or proxy the request to the target cluster.
+	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		proxyRequestInfo := lifted.NewRequestInfo(&http.Request{
+			URL:    &url.URL{Path: proxyPath, RawQuery: req.URL.RawQuery},
+			Method: req.Method,
+		})
+
+		requester, exist := request.UserFrom(req.Context())
+		if !exist {
+			responsewriters.InternalError(rw, req, errors.New("no user found for request"))
+			return
+		}
+		if proxyRequestInfo.Verb != "list" {
+			http.Error(rw, fmt.Sprintf("Request verb %s is not support", proxyRequestInfo.Verb), http.StatusMethodNotAllowed)
+			return
+		}
+		locker := sync.Mutex{}
+		wg := sync.WaitGroup{}
+		targetClusters := make([]requestContext, 0)
+		for i := range clusterList.Items {
+			wg.Add(1)
+			cluster := clusterList.Items[i]
+			go func(cluster *clusterapis.Cluster) {
+				defer wg.Done()
+				// TODO: store it in the memory
```

Comment:
Any idea for this? It's easy to store it in memory, but hard to keep it updated.

Reply:
Yes, so I don't have a good solution at the moment.

Reply:
Do you think this is a performance bottleneck? The whole operation doesn't have any slow IO operations(like a real HTTP request), since the required secret is obtained from cache.

Reply:
https://github.com/karmada-io/karmada/pull/4254/files#diff-00d6ef381b4e64b32856501a8161f47d60258491260bf5ddfa555dd980c968d6R67

There is a certain probability that a real HTTP request  will be sent.

Reply:
Yes, low probalility.
This concern can't be eliminated even by introducing another memory cache, right? For example, we might not be sure if the memory cache has loaded.

Reply:
Yes. 

So, are you suggesting to remove this TODO?

Reply:
Yeah, I agree.

Reply:
There is a similar TODO below, just a reminder.